ndb/src/google/cloud/ndb/model.py

```diff
@@ -487,6 +489,202 @@ def _verify_validator(validator):
 
         return validator
 
+    def __repr__(self):
+        """Return a compact unambiguous string representation of a property.
+
+        This cycles through all stored attributes and displays the ones that
+        differ from the default values.
+        """
+        args = []
+        cls = self.__class__
+        signature = inspect.signature(self.__init__)
+        for name, parameter in signature.parameters.items():
+            attr = "_{}".format(name)
+            instance_val = getattr(self, attr)
+            default_val = getattr(cls, attr)
+
+            if instance_val is not default_val:
+                if isinstance(instance_val, type):
+                    as_str = instance_val.__name__
```

Comment:
Maybe `__qualname__` would be better for types?

Reply:
1. I have no opposition to that change
1. Do you think that'd be a "backwards incompatible" change?

Reply:
The only thing I can think we'd be breaking might be doctests which use the `__repr__` result in place of assertions.

FWIW, PEP 3155 states, 

> For top-level functions and classes, the `__qualname__` attribute is equal to the `__name__` attribute

which means the only visible difference would be for nested classes.  How likely is it that class derived from `Property` has a `type` in its argument list, *and* that it is nested?

Reply:
I added `__qualname__`